contracts/UBI.sol

```diff
@@ -118,25 +151,53 @@ contract UBI is Initializable {
   */
   function startAccruing(address _human) external {
     require(proofOfHumanity.isRegistered(_human), "The submission is not registered in Proof Of Humanity.");
-    require(accruedSince[_human] == 0, "The submission is already accruing UBI.");
-    accruedSince[_human] = block.timestamp;
+    require(!activated[_human], "The submission is already accruing UBI.");
+    uint256 _now = block.timestamp;
+    if(stream[address(0)][_human] != 0) { // check incoming streams
+      balance[_human] += (stream[address(0)][_human] * (_now - accruedSince[_human]));
+    }
+    activated[_human] = true;
+    accruedSince[_human] = _now;
+    emit NewStream(address(0), _human, accruedPerSecond); // emit stream event 
+    // update M0 total supply marker 
+    supplyMarker += ((activeVerifiedHumans * accruedPerSecond) * (_now - lastSupplyUpdate));
+    activeVerifiedHumans++;
+    lastSupplyUpdate = _now;
   }
 
   /** @dev Allows anyone to report a submission that
   *  should no longer receive UBI due to removal from the
   *  Proof Of Humanity registry. The reporter receives any
   *  leftover accrued UBI.
   *  @param _human The submission ID.
+  *  @param _outFlow Array of outgoing stream addresses to revoke
   */
-  function reportRemoval(address _human) external  {
+  function reportRemoval(address _human, address[] calldata _outFlow) external  {
```

Comment:
Forcing the caller to manage the list of outflows seems like a disincentive for people to report removals. Could the list be managed by the contract internally?

Reply:
Callers have to grab that data from stream events. Outflows have limits per PoH so we don't have to manage a second/long list of address on-chain.. It's easy to implement but will cost more gas while creating new streams.